src/main/java/org/example/controllers/ClientController.java

```diff
@@ -0,0 +1,123 @@
+package org.example.controllers;
+
+import io.swagger.annotations.Api;
+import org.example.exceptions.DoesNotExistException;
+import org.example.exceptions.FailedToCreateException;
+import org.example.exceptions.InvalidException;
+import org.example.models.ClientRequest;
+import org.example.services.ClientService;
+
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.sql.SQLException;
+
+@Api("Client API")
+@Path("/api/client")
+public class ClientController {
+    private final ClientService clientService;
+
+    public ClientController(
+            final ClientService clientService
+    ) {
+        this.clientService = clientService;
+    }
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getClients() {
+        try {
+            return Response.ok().entity(
+                    clientService.getAllClients()
+            ).build();
+        } catch (SQLException e) {
+            return Response.status(
+                    Response.Status.INTERNAL_SERVER_ERROR
+            ).build();
+        }
+    }
+
+    @GET
+    @Path("/{id}")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getClientById(final @PathParam("id") int id) {
+        try {
+            return Response.ok().entity(
+                    clientService.getClientById(id)
+            ).build();
+        } catch (SQLException e) {
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR)
+                    .entity(e.getMessage()).build();
+        } catch (DoesNotExistException e) {
+            return Response.status(Response.Status.NOT_FOUND)
+                    .entity(e.getMessage()).build();
+        }
+
+    }
+
+    @POST
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response createClient(
+            final ClientRequest clientRequest) {
+        try {
+            return Response.ok()
```

Comment:
this could be a 201 instead as its created?

Reply:
this isnt a 201

Reply:
If you look below, ive put '.status(Response.Status.CREATED)', which returns 201?

Reply:
Line 70